docs/apigateway.md

```diff
@@ -1,53 +1,100 @@
-Follow the instructions in [Configure CLI](./README.md#setting-up-the-openwhisk-cli) on how to set the authentication key for your specific namespace.
-
-
-## Expose an OpenWhisk action
-
-Let's expose a simple action that is already pre-installed with OpenWhisk
-
-```
-$ wsk api-experimental create /hello /echo get /whisk.system/utils/echo
-```
-```
-ok: created api /echo GET for action /whisk.system/utils/echo
-https://21ef035.api-gw.mybluemix.net/hello/echo
-```
-A new URL is generated exposing the `echo` action via a __GET__ HTTP method.
 
-Let's give it a try by sending a HTTP request to the URL.
-```
-$ curl https://21ef035.api-gw.mybluemix.net/hello/echo?marco=polo
-```
-This will invoke the `echo` action, returning back a JSON string with the parameters sent.
-```
-{
-  "marco":"polo"
-}
-```
-
-You can pass parameters to the action via simple query parameters, or via request body.
+Follow the instructions in [Configure CLI](./README.md#setting-up-the-openwhisk-cli) on how to set the authentication key for your specific namespace.
 
-### Exposing multiple actions
+## Expose an OpenWhisk web action
+
+1. Create a JavaScript file with the following content. For this example, the file name is 'hello.js'.
+  ```javascript
+  function main({name:name='Serverless API'}) {
+      return {payload: `Hello world ${name}`};
+  }
+  ```
+  
+2. Create a web action from the following JavaScript function. For this example, the action is called 'hello'. Make sure to add the flag `--web true`
+  
+  ```
+  wsk action update hello hello.js --web true
+  ```
+  ```
+  ok: updated action hello
+  ```
+  
+3. Create an API with base path `/hello`, path `/world` and method `get` with response type `json`
+  
+  ```
+  wsk api create /hello /world get hello --response-type json
+  ```
+  ```
+  ok: created API /hello/world GET for action /_/hello
+  https://${APIHOST}:9001/api/21ef035/hello/world
+  ```
+  A new URL is generated exposing the `hello` action via a __GET__ HTTP method.
+  
+4. Let's give it a try by sending a HTTP request to the URL.
+  
+  ```
+  $ curl https://${APIHOST}:9001/api/21ef035/hello/world?name=OpenWhisk
+  ```
+  ```json
+  {
+  "payload": "Hello world OpenWhisk"
+  }
+  ```
+  The action `hello` got invoked, returning back a JSON string including the parameter `name` sent via query parameter. You can pass parameters to the action via simple query parameters, or via request body.
+  
+5. Full controll over the the http response
+  
+  Notice that the full result of the action is returned in the body of the response, and the content is assume to be `application/json` this is because the API was created with `--response-type json` using `json` for the response-type allows you to easily get started with existing actions, but once you have something working you want to have full control over the http response like `statusCode`, `headers` and return different content types in the `body`.
```

Comment:
Very long single sentence. How about?

The `--response-type` flag controls the the output of the HTTP response from the API Gateway. Using `--response-type json` as above returns the full result of the action in JSON format and automatically sets the Content-Type header to `application/json` which enables you to easily get started.  In addition to automatic JSON support, API Gateway allows you to fully control the HTTP response's status code, HTTP headers and content type of the body.